src/components/general/DatePicker.js

```diff
@@ -3,195 +3,132 @@ import PropTypes from "prop-types";
 import { View } from "react-native";
 import { Input } from "./Input";
 import { PickerInput } from "./PickerInput";
+import { prepend0 } from "../../utils/events";
 
 class DatePicker extends Component {
 	constructor(props) {
 		super(props);
 
 		let date = [];
 		if (this.props.value) date = this.props.value.split("-");
-
+		console.log(date);
 		this.state = {
 			month: date.length === 3 ? date[1] : "",
 			day: date.length === 3 ? date[2] : "",
 			year: date.length === 3 ? date[0] : "",
-			monthArr: Array.from({ length: 12 }, (v, k) => k + 1),
-			dayArr: Array.from({ length: 31 }, (v, k) => k + 1),
-			yearArr: Array.from({ length: 101 }, (v, k) => new Date().getFullYear() - k),
-			focused: date.length === 3
+			monthArray: Array.from({ length: 12 }, (v, k) => k + 1),
+			yearArray: Array.from({ length: 20 }, (v, k) => new Date().getFullYear() + k) // We only need about 20 years ahead... right?
 		};
 	}
 
-		static propTypes = {
-			value: PropTypes.object,
-			placeholder: PropTypes.string.isRequired,
-			onSelect: PropTypes.func.isRequired
-		}
-
-		prepend0(item) {
-			if (item < 10)
-				return "0" + item;
-
-			return item;
-		}
-
-		leapYear(year) {
-			return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
-		}
+	static propTypes = {
+		value: PropTypes.object,
+		onSelect: PropTypes.func.isRequired
```

Comment:
There should be a placeholder to tell the user what the field in question is asking for.